docs/content/sips/006-spin-plugins.md

```diff
@@ -0,0 +1,233 @@
+- The binary of the plugin must be named `$name`
+- The latest plugin manifest file must be named `$name.json`
+- The license for the plugin must be named `$name.license`
+
+#### Plugin Compatibility
+
+Spin plugins must specify compatible versions of Spin in the `spinCompatibility` field of the manifest. The field is expected to be a [comparison operator](https://docs.rs/semver/1.0.13/semver/enum.Op.html) (`=, >, >=, <, <=, ~, ^, *`) along with the compatible version of Spin. The JSON schema validates that the `spinCompatibility` field is a string that matches the following regular expression: `^[><~^*]?[=]?v?(0|[1-9]\d*)(\.(0|[1-9]\d*))?(\.(0|[1-9]\d*))?(-(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*)?(\+[0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*)?$`. 
+For example, specifying `=0.4` means that the plugin is compatible with versions equivalent to `>=0.4.0, <0.5.0`.
+
+Spin will use the [`semver`](https://docs.rs/semver/1.0.13/semver/struct.VersionReq.html) crate that inspired this syntax to verify that the plugin works on the current version of Spin. If it does not, it will fail to install the plugin and log a message explaining the version mismatch. An `--skip-version-compat-check` flag can be added to a plugin installation to ignore this error.
+
+#### Centralized Plugin Manifest Repository
+
+- A new GitHub repository https://github.com/fermyon/spin-plugins will act as the index for all the Spin plugin manifests. Having a centralized location for plugin manifests enables future support of a `spin plugin search` subcommand that allows users to search for plugins via the Spin CLI.
+- Creators of new plugins can submit PRs to add a plugin manifest to the repository.
+- Plugin creators are required to test Spin compatibility with their plugin and update the `spinCompatability` field of the manifest over time accordingly.
+- Plugin manifests can be hosted elsewhere and installed via the `--file` or `--url` fields of `spin plugin install`.
+
+## Future design considerations
+
+### Larger scope for Spin Plugins
+
+The concept of Spin plugins is to allow both new subcommands and functionality to be added to Spin. This SIP focuses on the former, enabling users to both install and execute subcommands from the Spin CLI; however, there are cases where it may be useful to install a new Spin feature that is executed by Spin rather than the user. An example of this is Spin triggers. A user may wish to [extend Spin to support a timer trigger](https://spin.fermyon.dev/extending-and-embedding/) that executes components at a configured time interval. Instead of having to understand, modify, and grow the spin codebase, a user could package the trigger as a plugin. After installing the trigger via `spin plugin install`. Spin could invoke it when a Spin manifest references the trigger. 
+
+While for now plugins are assumed to be executables, in the future, support for plugging in WebAssembly modules may be desirable.
```

Comment:
Should this be a separate topic, i.e. `### WebAssembly Plugin Support`?

Reply:
added in 24adf4a8de2a7627d